internal/handlers/pg/msg_renamecollection.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pg
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgRenameCollection implements HandlerInterface.
+func (h *Handler) MsgRenameCollection(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	var err error
+	dbPool, err := h.DBPool(ctx)
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.Unimplemented(document, "writeConcern", "comment"); err != nil {
+		return nil, err
+	}
+
+	var dropTarget bool
+
+	if dropTarget, err = common.GetOptionalParam(document, "dropTarget", false); err != nil {
+		return nil, err
```

Comment:
I don't see codecov, but I suspect this err is not tested, we should test it. If dropTarget was not boolean.

Reply:
It will return `ErrTypeMismatch` if `dropTarget` is not a boolean. It asserts on the key `dropTarget` if of type T [here](https://github.com/FerretDB/FerretDB/blob/v1.0.0/internal/handlers/common/params.go#L54). Please correct me if I am mistaken.

Reply:
We should also handle an invalid value for the option too. I will add that.

Reply:
I have removed `dropTarget` from scope.